src/components/widget/ShortenField.jsx

```diff
@@ -0,0 +1,38 @@
+import style from './ShortenField.module.css';
+import { Button } from '../elements/Button';
+import { useState, useRef } from 'react';
+
+function ShortenField() {
+  const [showError, setShowError] = useState(false);
+  const inputRef = useRef(null);
+
+  const handleShort = () => {
+    const urlPattern =
+      /[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/gi;
+    const regex = new RegExp(urlPattern);
+    const inputValue = inputRef.current.value.trim();
+    if (inputValue === '' || !inputValue.match(regex)) {
+      setShowError(true);
+    } else {
+      setShowError(false);
+    }
+  };
+  return (
+    <div className={style.field_block}>
+      <input
+        ref={inputRef}
+        className={`${style.field} ${showError ? style.invalid : ''}`}
+        type="text"
```

Comment:
It might be type `url`, but it's up to you.

Reply:
The problem here is than I make url type I can't use the function onSubmit and change setShowError because input makes a message before

Reply:
It's a native behaviour. I guess there is a way to disable it if you like.